app/models/manageiq/providers/inventory/persister/builder/storage_manager.rb

```diff
@@ -23,7 +23,6 @@ def cloud_object_store_containers
         end
 
         def host_initiators
-          add_properties(:parent_inventory_collections => [:physical_storages])
```

Comment:
So a little background on why this is wrong.

These parent_inventory_collections are intended to be used only when the collection has no direct relation to the parent manager (aka ext_management_system).  An example of this would be disks which belong to a hardware record not to the ext_management_system.

When this is the case the parent_inventory_collection is used to determine the delete_scope, e.g. when a vm is the target then the parent_inventory_collection of `:vms` is used to find all existing disks in that scope.

HostInitiators belongs_to an ext_management_system and so this isn't necessary and is causing targeted refresh failures.

Reply:
Great to know. I spend hours trying to find the problem. I will pass this info to the rest of the team. thanks

Reply:
I checked it and indeed this solved the issue.

Reply:
:+1: Thanks @llivne